src/index.ts

```diff
@@ -410,18 +392,25 @@ export function extractValueOrLiteral<
 export type Rule =
   | string
   | {
-    if: Rule
+    if: And | Or | string
     then: Rule
     else?: Rule
   }
   | {
-    and: Rule[]
+    and: And
```

Comment:
I think this might be doubling the `{and: {and` bit.

Reply:
"might be" 🤣 You're too kind.